x-pack/plugins/snapshot_restore/server/routes/api/repositories.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const getAll = {
+  path: 'repositories',
+  method: 'get',
+  handler: async (req: any, callWithRequest: any, responseToolkit: any) => {
```

Comment:
From looking at how some other plugins do it, I think we can use a few types here:

```js
import { Legacy } from 'kibana';
import { Request, ResponseToolkit } from 'hapi';
import { CallClusterWithRequest } from 'src/legacy/core_plugins/elasticsearch';

handler: async (req: Legacy.Request, callWithRequest: CallClusterWithRequest, responseToolkit: ResponseToolkit) => {

// or this one... not sure why we have both Legacy.Request and Request
handler: async (req:Request, callWithRequest: CallClusterWithRequest, responseToolkit: ResponseToolkit) => {
```